metaflow/plugins/aws/aws_client.py

```diff
@@ -45,12 +47,26 @@ def get_client(module, with_error=False, params={}):
             return boto3.session.Session(**cached_aws_sandbox_creds).client(
                 module, **params
             )
+        session = boto3.session.Session()
+        if role:
+            fetcher = botocore.credentials.AssumeRoleCredentialFetcher(
+                client_creator=session._session.create_client,
+                source_credentials=session._session.get_credentials(),
+                role_arn=role,
+                extra_args={},
+            )
+            creds = botocore.credentials.DeferredRefreshableCredentials(
+                method="assume-role", refresh_using=fetcher.fetch_credentials
+            )
+            botocore_session = botocore.session.Session()
+            botocore_session._credentials = creds
+            session = boto3.session.Session(botocore_session=botocore_session)
```

Comment:
Would this still work if original creds are temporary creds from IMDS? (I _think_ it would but not 100% sure)